Source/Engine/PackageBuilder/SyntaxParser.cs

```diff
@@ -1345,6 +1404,12 @@ private void ValidateToken(TokenId id, string error)
                 throw SyntaxError(fToken.TextSlice.Position, error, args: fToken);
         }
 
+        private void AttachSourceTextInformation(Syntax syntax, int start)
+        {
+            int end = fLastToken.TextSlice.Position + fLastToken.TextSlice.Length;
+            syntax.TextRange = new TextRange(start, end);
+        }
+
```

Comment:
Seems fLastToken is not needed. You may use fToken.TextSlice.Position to get the end of the parsed syntax node. Note that End in TextRange is exclusive, i.e. it points to the position after the syntax node (the length is calculated as Length = End - Start).

Reply:
Actually, beginning of the next token is not equal to the end of previous since there can be any number of spaces between them. So TextRange would include not only tokens of the syntax but all the consequent spaces.